customArguments_test.go

```diff
@@ -285,146 +222,125 @@ var _ = Describe("customArguments", func() {
 		It("returns an existing onCall object if one exists for that index", func() {
 			result := ca.OnCall(2)
 
-			o, ok := result.(*onCall)
-
 			Expect(ca.onCalls).To(HaveLen(3))
-			Expect(ok).To(BeTrue())
-			Expect(*o).To(Equal(onCall{stub: mockFn, index: 2}))
+			Expect(result).To(Equal(&OnCall{stub: stub, index: 2}))
 		})
 	})
 
 	Describe("OnFirstCall", func() {
 		It("creates a new onCall with a 0 index", func() {
-			ca := &customArguments{
-				stub: mockFn,
-				onCalls: []*onCall{
-					&onCall{stub: mockFn, index: 1},
-					&onCall{stub: mockFn, index: 2},
+			ca := &CustomArguments{
+				stub: stub,
+				onCalls: []*OnCall{
+					&OnCall{stub: stub, index: 1},
+					&OnCall{stub: stub, index: 2},
 				},
 			}
 
 			result := ca.OnFirstCall()
 
-			o, ok := result.(*onCall)
-
 			Expect(ca.onCalls).To(HaveLen(3))
-			Expect(ok).To(BeTrue())
-			Expect(*o).To(Equal(onCall{stub: mockFn, index: 0}))
+			Expect(result).To(Equal(&OnCall{stub: stub, index: 0}))
 		})
 
 		It("returns the existing first call", func() {
-			ca := &customArguments{
-				stub: mockFn,
-				onCalls: []*onCall{
-					&onCall{stub: mockFn, index: 1},
-					&onCall{stub: mockFn, index: 2},
-					&onCall{stub: mockFn, index: 0},
+			ca := &CustomArguments{
+				stub: stub,
+				onCalls: []*OnCall{
+					&OnCall{stub: stub, index: 1},
+					&OnCall{stub: stub, index: 2},
+					&OnCall{stub: stub, index: 0},
 				},
 			}
 
 			result := ca.OnFirstCall()
 
-			o, ok := result.(*onCall)
-
 			Expect(ca.onCalls).To(HaveLen(3))
-			Expect(ok).To(BeTrue())
-			Expect(*o).To(Equal(onCall{stub: mockFn, index: 0}))
+			Expect(result).To(Equal(&OnCall{stub: stub, index: 0}))
 		})
 	})
 
 	Describe("OnSecondCall", func() {
 		It("creates a new onCall with a 1 index", func() {
-			ca := &customArguments{
-				stub: mockFn,
-				onCalls: []*onCall{
-					&onCall{stub: mockFn, index: 0},
-					&onCall{stub: mockFn, index: 2},
+			ca := &CustomArguments{
+				stub: stub,
+				onCalls: []*OnCall{
+					&OnCall{stub: stub, index: 0},
+					&OnCall{stub: stub, index: 2},
 				},
 			}
 
 			result := ca.OnSecondCall()
 
-			o, ok := result.(*onCall)
-
 			Expect(ca.onCalls).To(HaveLen(3))
-			Expect(ok).To(BeTrue())
-			Expect(*o).To(Equal(onCall{stub: mockFn, index: 1}))
+			Expect(result).To(Equal(&OnCall{stub: stub, index: 1}))
```

Comment:
is the intention for the values to match and not the addresses? I think ginkgo is using the underlying values anyway. Might be more readable to do `Expect(*result).To(Equal(OnCall{stub: stub, index: 1}))`